src/praisonai-agents/praisonaiagents/agent/chat_mixin.py

```
@@ -699,11 +700,19 @@ def _chat_completion(self, messages, temperature=1.0, tools=None, stream=True, r
+                        return self._chat_completion(
+                            truncated_messages, temperature, tools, stream, 
+                            reasoning_steps, task_name, task_description, task_id, response_format, 
+                            _retry_depth=_retry_depth + 1
+                        )
+                    else:
+                        logging.error(f"[{self.name}] Context overflow retry limit exceeded")
+                        raise LLMError(
+                            f"Context overflow could not be resolved after {_retry_depth} attempts", 
+                            model_name=model_name, agent_id=self.name, is_retryable=False
+                        ) from e
                 except Exception as recovery_error:
```

Ruff (0.15.12)

[warning] Do not catch blind exception: `Exception`

(BLE001)